src/apcb.rs

```diff
@@ -293,41 +277,34 @@ impl<'a> ApcbIterMut<'a> {
             let group_size = group.header.group_size.get();
             if group.header.group_id.get() == group_id {
                 return Ok((offset, group_size as usize));
-            } else {
-                let group = ApcbIterMut::next_item(&mut self.buf)?;
-                let group_size = group.header.group_size.get() as usize;
-                offset = offset
-                    .checked_add(group_size)
-                    .ok_or(Error::ArithmeticOverflow)?;
-                remaining_used_size = remaining_used_size
-                    .checked_sub(group_size)
-                    .ok_or(Error::FileSystem(
-                        FileSystemError::InconsistentHeader,
-                        "GROUP_HEADER::group_size",
-                    ))?;
             }
+            let group = ApcbIterMut::next_item(&mut self.buf)?;
+            let group_size = group.header.group_size.get() as usize;
+            offset = offset
+                .checked_add(group_size)
+                .ok_or(Error::ArithmeticOverflow)?;
+            remaining_used_size = remaining_used_size
+                .checked_sub(group_size)
+                .ok_or(Error::FileSystem(
+                FileSystemError::InconsistentHeader,
+                "GROUP_HEADER::group_size",
+            ))?;
         }
         Err(Error::GroupNotFound)
     }
 
     pub(crate) fn next1(&mut self) -> Result<GroupMutItem<'a>> {
-        if self.remaining_used_size == 0 {
-            panic!("Internal error");
-        }
-        match Self::next_item(&mut self.buf) {
-            Ok(e) => {
-                let group_size = e.header.group_size.get() as usize;
-                if self.remaining_used_size >= group_size {
-                } else {
-                    return Err(Error::FileSystem(
-                        FileSystemError::InconsistentHeader,
-                        "GROUP_HEADER::group_size",
-                    ));
-                }
-                self.remaining_used_size -= group_size;
-                Ok(e)
-            }
-            Err(e) => Err(e),
+        assert!(self.remaining_used_size != 0, "Internal error");
```

Comment:
Can these asserts be disabled? I would not like to corrupt anything.

Reply:
No.  They run in both debug and release modes, unconditionally.

From https://doc.rust-lang.org/std/macro.assert.html:
> Assertions are always checked in both debug and release builds, and cannot be disabled. See [debug_assert!](https://doc.rust-lang.org/std/macro.debug_assert.html) for assertions that are not enabled in release builds by default.